### PR TITLE
Wrap output events in container with step and action UUIDs

### PR DESCRIPTION
**Pull request by rowanseymour**

### CODE REVIEW

File: cmd/docgen/run.go [hunk 1]
```diff
@@ -91,10 +91,10 @@ func eventsForAction(actionJSON []byte) (json.RawMessage, error) {
 		return nil, err
 	}
 
-	events := session.Events()
-	eventJSON := make([]json.RawMessage, len(events))
-	for i, event := range events {
-		typed, err := utils.EnvelopeFromTyped(event)
+	eventLog := session.EventLog()
```
Comment (rowanseymour) [inline cmd/docgen/run.go]: Definitely open to naming discussion - we need terminology that makes it clear that these are not events but something else that contains an event and information about how it was generated

File: cmd/flowrunner/main.go [hunk 2]
```diff
@@ -21,8 +21,8 @@ import (
 )
 
 type Output struct {
-	Session json.RawMessage        `json:"session"`
-	Events  []*utils.TypedEnvelope `json:"events"`
+	Session  json.RawMessage   `json:"session"`
+	EventLog []json.RawMessage `json:"events"`
```
Comment (rowanseymour) [inline cmd/flowrunner/main.go]: Haven't done it yet but I think this should now be something other than just `events` for clarity - e.g. `event_log` or `log`

File: cmd/flowrunner/testdata/flows/all_actions_test.json [hunk 3]
```diff
@@ -1,4 +1,5 @@
 {
+  "caller_events": [],
```
Comment (rowanseymour) [inline cmd/flowrunner/testdata/flows/all_actions_test.json]: Randomness has decided that "caller_events" should always come first in the test files now

File: cmd/docgen/run.go (hunk 1 above)
Comment (ericnewcomer) [inline cmd/docgen/run.go]: How about just output?

File: cmd/flowrunner/main.go (hunk 2 above)
Comment (ericnewcomer) [inline cmd/flowrunner/main.go]: Hmmm.. guess we have an Output already. Maybe event_log makes sense then.

File: cmd/docgen/run.go (hunk 1 above)
Comment (rowanseymour) [inline cmd/docgen/run.go]: I like output but what would be the name of each item in output?

Comment (ericnewcomer) [inline cmd/docgen/run.go]: Could be OutputLog, but Output having output with output logs? That's a bit nutty. Event log at least removes confusion with top level Output.

File: cmd/flowrunner/testdata/flows/all_actions_test.json (hunk 3 above)
Comment (ericnewcomer) [inline cmd/flowrunner/testdata/flows/all_actions_test.json]: or the alphabet?

Comment (rowanseymour) [inline cmd/flowrunner/testdata/flows/all_actions_test.json]: ah i forgot they were previously called "resume_events" before I renamed them to "caller_events"

File: flows/runs/event_log.go [hunk 4]
```diff
@@ -0,0 +1,56 @@
+package runs
+
+import (
+	"encoding/json"
+
+	"github.com/nyaruka/goflow/flows"
+	"github.com/nyaruka/goflow/utils"
+)
+
+type eventLogEntry struct {
```
Comment (nicpottier) [inline flows/runs/event_log.go]: This naming seems a bit tortured. It is a log, of events, but actually no it is a log of entries which are of events? 

What about it is just a log and it has log entries? We don't need to describe that the log entries are of events do we?

File: cmd/flowrunner/testdata/flows/all_actions_test.json (hunk 3 above)
Comment (nicpottier) [inline cmd/flowrunner/testdata/flows/all_actions_test.json]: Looks like while golang does have random map iteration, json encoding of maps is sorted by key.
https://golang.org/src/encoding/json/encode.go

File: flows/runs/event_log.go (hunk 4 above)
Comment (rowanseymour) [inline flows/runs/event_log.go]: so `session.Log() []flows.LogEntry`  ? It's a bit vague but I like the brevity

how do we feel about using `log` in response, e.g. 

```json
{
  "session": {},
  "log": [{}, {}]
}
```

Comment (nicpottier) [inline flows/runs/event_log.go]: I like log in response. And ya I like the brevity as well.

**Pull request merged**